### ISSUE TITLE
doughwallet

### ISSUE THREAD

**Issue by maerco** [iancoleman/bip39#64]
hi, I was looking a way to use your tool with the iphone doge coin app [doughwallet](https://github.com/peritus/doughwallet).

I'm not able to get the proper derivation path to get the correct adresses.

While I was able to use it successfully with the dogecoin app [breadwallet](https://github.com/voisine/breadwallet), from which dough has been fork, using the derivation path m/0'/0/*.

Do you have any Idea which is the correct derivation path for this app?

**Comment by iancoleman** [comment]
I do not know what the correct derivation path is for doughwallet.

Can you please supply some info for doge and doughwallet containing

* the mnemonic
* the dogecoin address generated by doughwallet for that mnemonic

[Related issue](https://github.com/peritus/doughwallet/issues/6)

**Comment by maerco** [comment]
Here a new dogecoin address

DJu2zHGLbQDiVPMxvPMZ52YAgUrEwifuBZ

Generated with following phrase:

<Code>Until clog assist play slide hybrid return rubber possible undo item together</code>

**Comment by iancoleman** [comment]
This is an issue with doughwallet.

After a lot of looking, I can't find where the 'wrong' code is in doughwallet (see attached for diff with breadwallet before doge changes). I'm fairly confident it's something that *hasn't* been changed that should have, rather than an incorrect change made by the author of doughwallet.

The derivation path seem to have been kept the same as breadwallet, so should be set to m/0'/0 in the bip32 tab of this tool.

The following alternative tools confirm the phrase and bip32 seed above should generate a first address of `DEPLJffgPBuLqHUA24gCuKJmnp89f6ui5k`, not the one
[doughwallet.txt](https://github.com/iancoleman/bip39/files/950221/doughwallet.txt)
 doughwallet generates.

https://bip32jp.github.io/english/index.html
http://bip32.org/

I confirmed the same phrase used in breadwallet generates matching addresses across all tools and wallets, so the fork into doughwallet should be ok if the modifications to doge are correct (although doughwallet is 1039 commits behind breadwallet so take it with a grain of salt).

My recommendation: stop using doughwallet.

**Comment by maerco** [comment]
Thanks for your analysis Ian.

**Comment by choas** [comment]
I've run DoughWallet in an iOS simulator and found the private key in the debugger:
QPLtCjDJ89eYGpEPPbhnZZyydhLt1ywiC3sCaXr6aEUCcSxygbWz

For more information see https://www.reddit.com/r/dogecoin/comments/6qicz9/how_to_get_your_private_keys_out_of_doughwallet/

tl;dr
BRBIP32Sequence.m line 251

**Comment by bnyk25** [comment]
Hello Ian,

Dough Wallet now has recovery tool, with derivation path set to m/0'/0. For wallets who had outgoing transactions, changing the path to m/0'/1 generates the private keys. But for wallets who had no outgoing transactions, the path m/0'/0 does not seem to work.

Is it possible for the child path to be m/0'/1 while the parent path not be m/0'/0?

Thank you.

**Comment by cyclone-github** [comment]
Due to the lack of info available on the custom Dough Wallet derivation or how to recover these wallets, I published my research and a custom tool:
https://github.com/cyclone-github/doughwallet_recovery
